jaqpotpy/jaqpot.py

```diff
@@ -111,6 +115,178 @@ def set_api_key(self, api_key):
+        organization_id : organization id
+
+        """
+        jaqpot_api_client = JaqpotApiClient(
+            host=self.api_url, access_token=self.api_key
+        )
+        model_api = ModelApi(jaqpot_api_client)
+        response = model_api.get_shared_models_with_http_info(
+            page=page, size=size, sort=sort, organization_id=organization_id
+        )
+        if response.status_code < 300:
+            shared_models = response.data
+            data = []
+            for shared_model in shared_models.content:
+                model_info = {
+                    "Name": shared_model.name,
+                    "model_id": shared_model.id,
+                    "Type": shared_model.type,
+                    "shared_with_organizations_id": shared_model.shared_with_organizations,
+                }
+                data.append(model_info)
+            df = pd.DataFrame(data)
+            return df
+        else:
+            self.log.error("Error code: " + str(response.status_code.value))
+
+    def get_dataset_by_id(self, dataset_id):
+        """Get dataset from Jaqpot.
+
+        Parameters
+        ----------
+        dataset_id : dataset_id is the id of the dataset on Jaqpot
+
+        """
+        jaqpot_api_client = JaqpotApiClient(
+            host=self.api_url, access_token=self.api_key
+        )
+        dataset_api = DatasetApi(jaqpot_api_client)
+        response = dataset_api.get_dataset_by_id_with_http_info(id=dataset_id)
+        if response.status_code < 300:
+            dataset = response.data
+            return dataset
+        else:
+            self.log.error("Error code: " + str(response.status_code.value))
+
+    def predict_with_model(self, model_id, dataset):
+        """Predict with model on Jaqpot.
+
+        Parameters
+        ----------
+        model_id : model_id is the id of the model on Jaqpot
+        dataset : dataset to predict
+
+        """
+        dataset = Dataset(
+            type=DatasetType.PREDICTION,
+            entry_type="ARRAY",
+            input=dataset,
+        )
+        jaqpot_api_client = JaqpotApiClient(
+            host=self.api_url, access_token=self.api_key
+        )
+        model_api = ModelApi(jaqpot_api_client)
+        response = model_api.predict_with_model_with_http_info(
+            model_id=model_id, dataset=dataset
+        )
+        if response.status_code < 300:
+            dataset_location = response.headers["Location"]
+            dataset_id = int(dataset_location.split("/")[-1])
+            completed_prediction = False
+            while not completed_prediction:
+                dataset = self.get_dataset_by_id(dataset_id)
+                if dataset.status == "SUCCESS":
+                    completed_prediction = True
+                    prediction = dataset.result
+                    return prediction
+                elif dataset.status == "FAILURE":
+                    self.log.error("Prediction failed")
+                    return
```

Comment:
Better throw like this:
```python
raise PredictionError(f"Prediction failed: {str(e)}")
```
instead of returning, so the user knows that smth failed and can use try.. catch when calling the predict method and handle it appropriately.

Reply:
I'm not sure i get this one. Do you mean replacing 

`self.log.error("Prediction failed")
  return`
  
  with 
  
  `raise Exception("Prediction failed")`?

Reply:
yes

Reply:
Actually not a generic `RuntimeError`. A `JaqpotPredictionException` would be more appropriate, so the user can handle it when calling it like this:
```python
    try:
        makePrediction()
    except JaqpotPredictionError:
        # handle the error here if the prediction failed
    except JaqpotPredictionTimeoutError:
        # re-send a new prediction etc. 
    except RuntimeError: 
        # handle random runtime error 
```